src/PSR7/SpecFinder.php

```diff
@@ -262,6 +266,8 @@ public function findHeaderSpecs(OperationAddress $addr): array
         // 2. Collect path-level headers from "parameters" keyword
         // Path level params are fall-backs
         $pathSpec = $this->findPathSpec($addr);
+
+        /** @var Parameter $p */
```

Comment:
This can't be inferred?

Reply:
`$pathSpec->parameters` has type of `Parameter[]|Reference[]` So it PHPStan thinks it's `Parameter|Reference` So we need to force it to think it's just parameter.

Reply:
does it work because we forced to resolve references beforehand? just to confirm it's not a potential bug (which is why phpstan actually comes here)

Reply:
I'm not sure. Maybe we don't have a test scenario that has headers with reference.

Reply:
If we're not sure we can drop this one into baseline ignores too. I just don't want to hardcode things we are not sure in.

Reply:
Ok, it's a kind of bug :smile: If you have a spec with header parameters with references and you use this the following code
```php
(new ValidatorBuilder())
    ->setSchemaFactory(new PrecreatedSchemaFactory(Reader::readFromYaml($yaml)))
    ->getServerRequestValidator()
    ->getSchema();
```

You get an error `Undefined property: cebe\openapi\spec\Reference::$in` at `src/PSR7/SpecFinder.php:250`

I would say this is an edge case. The majority of people are using JsonFactory or YamlFactory which resolves the references.

What do you think? Should I create a separate issue for this? Or we can ignore it?

Reply:
It would be nice to file an issue **and** ignore this with phpstan until it's resolved. so yes, phpstan actually found a bug. that's I was talking about :D
👍

Reply:
Ok. Reverted this. But funny PHPStan doesn't complain about it anymore. Weird..